feat (processor/k8sattributes): wait for synced when starting k8sattributes processor. (#32622)
open-telemetry/opentelemetry-collector-contrib: processor/k8sattributesprocessor/README.md

}
```

> By default, the processor will be ready as soon as it starts, even no metadata has been fetched yet.

Reviewer: ```suggestion
By default, the processor will be ready as soon as it starts, even if no metadata has been fetched yet.
```

If data is sent to this processor before the metadata is synced, there will be no metadata to enrich the data with.

To wait for the metadata to be synced before the processor is ready, set the `wait_for_metadata` option to `true`.

> Then the processor will not be ready until the metadata has been synced.

Reviewer: Does this mean that the Collector is blocked from getting started until the processor becomes ready?
If that's the case, maybe this should be mentioned explicitly here.

Author: ```suggestion
Then the processor will not be ready until the metadata is fully synced. As a result, the start-up of the Collector tool will be blocked. If the metadata cannot be synced, the Collector tool will ultimately fail to start.
```

Author: Yes. I'll add it.

> Then the processor will not be ready until the metadata is fully synced. As a result, the start-up of the Collector tool will be blocked. If the metadata cannot be synced, the Collector tool will ultimately fail to start.

Reviewer: ```suggestion
Then the processor will not be ready until the metadata is fully synced. As a result, the start-up of the Collector will be blocked. If the metadata cannot be synced, the Collector will ultimately fail to start.
```
